### PR TITLE
Arrange the interceptor in each handler instead of the framework list

### PR DESCRIPTION
**Pull request by dreamershl**

### CODE REVIEW

**Comment by jfarcand** [comment]
Thanks, that's a good start but one of the unit test is blocked forever. Looking at it.

**Comment by dreamershl** [comment]
Seems the reason is because the current test cases are constructed based on original idea, the interceptor will be added into the default interceptors. 

For example, the actionCancelledTest(), add the handler after the adding interceptor, the test will be OK.

I will modify all the fail test cases and submit the pull request again.

**Comment by dreamershl** [comment]
In order not to change the original add interceptor semantics, now the newly added interceptor will be also added into the existing handlers. So all the current test case are green now :)

Please have a check.

**Comment by slovdahl** [comment]
Could you please format the code as the rest of the project as well? The default style in IntelliJ IDEA is used. And no wildcard imports.

**Comment by dreamershl** [comment]
It is done. please check

File: modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereFramework.java [hunk 1]
```diff
@@ -986,7 +987,8 @@ private void info() {
             logger.info("Using BroadcastFilter: {}", i);
         }
 
-        if (broadcasterCacheClassName == null || DefaultBroadcasterCache.class.getName().equals(broadcasterCacheClassName)) {
+        if (broadcasterCacheClassName == null || DefaultBroadcasterCache.class.getName().equals(
+                broadcasterCacheClassName)) {
             logger.warn("No BroadcasterCache configured. Broadcasted message between client reconnection will be LOST. " +
```
Comment (jfarcand) [inline modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereFramework.java]: OK minor, but don't reformat as it pollute your original fix.

File: modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereFramework.java [hunk 2]
```diff
@@ -2839,7 +2898,9 @@ public AtmosphereFramework addWebSocketHandler(String path, WebSocketHandler han
     public AtmosphereFramework addWebSocketHandler(String path, WebSocketHandler handler, AtmosphereHandler h, List<AtmosphereInterceptor> l) {
         WebSocketProcessorFactory.getDefault().getWebSocketProcessor(this)
                 .registerWebSocketHandler(path,
-                        new WebSocketProcessor.WebSocketHandlerProxy(broadcasterFactory.lookup(path, true).getClass(), handler, interceptors));
+                        new WebSocketProcessor.WebSocketHandlerProxy(
+                                broadcasterFactory.lookup(path, true).getClass(),
+                                handler, interceptors));
```
Comment (jfarcand) [inline modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereFramework.java]: Don't reformat and don't add white line :-) Thanks a lot BTW!!

**Comment by jfarcand** [comment]
@dreamershl Sorry to be picky, but could you first reformat the class and then add your changes? There are still cosmetics changed mixed with your patch. For the patch, @elakito / @slovdahl what do you think?

**Comment by dreamershl** [comment]
Done. maybe next time, you can export your setting and share it to the contributor. The coding style will be aligned :). Thanks

**Comment by jfarcand** [comment]
@dreamershl I still see failures:
```java
Results :

Failed tests:
  ManagedAtmosphereHandlerTest.testPriority:346 expected:<XXX> but was:<@ManagedService Interceptor>



Tests run: 201, Failures: 1, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] atmosphere-project ................................. SUCCESS [  0.318 s]
[INFO] atmosphere-runtime ................................. FAILURE [ 24.091 s]
```

**Comment by dreamershl** [comment]
I only run the test cases in the cpr. From the following, the failure should be caused by the managed annotation process doesn't sort the interceptors.  Need some time to figure out how the framework interpret the annotations and add the sorting step accordingly.

image

**Comment by dreamershl** [comment]
done. it is because of the wrongly sorted sequence. And now all 196 test case are all passed.

image

**Comment by jfarcand** [comment]
@dreamershl OK there are regression with the [integration-tests/perftest](https://github.com/Atmosphere/integration-tests/tree/master/perf-tests). It seems now that interceptor are installed globally for all @Service defined.

**Comment by dreamershl** [comment]
Is that the purpose for the change? all the service should have a separate interceptors sequence.

**Comment by jfarcand** [comment]
@dreamershl Yes, but the wrong one get added, e.g. an annotated `WebSocketHandlerService` should not get in its `interceptors` list `AtmosphereHandlerInterceptor`. That's the issue currently.

**Comment by gdrouet** [comment]
@jfarcand there is a regression on interceptor installation with meta service (STOMP extension is using it) because of those changes: https://github.com/dreamershl/atmosphere/blob/dbad33a95bedad9374cd2ade58e9f2cd221571a3/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereFramework.java#L2537-L2542

interceptor is called by loadMetaService() and configureAtmosphereInterceptor() so the same interceptor could be registered in both calls in the AtmosphereHandlerWrapper. I think I could fix it by just looking for existing classes to make sure wrapper.interceptors.add(c); is not called many times for the same interceptor's class.

What do you think?

**Comment by jfarcand** [comment]
@gdrouet I agree. I'm curently fixing #1890 which also reproduce the same issue.

**Comment by jfarcand** [comment]
@gdrouet OK Can you test it? Since the stomp sample is broken, I cannot garantee it work. Let's continue the discussion undeer the 1905. Thanks!!!

**Pull request merged**